tempest/api/workloadmgr/snapshot/test_snapshots.py

```diff
@@ -210,14 +212,20 @@ def test_4_delete_snapshot(self):
                 reporting.add_test_step(
                     "Execute snapshot-delete command", tvaultconf.PASS)
                 LOG.debug("Command executed correctly")
+            start_time = time.time()
             time.sleep(5)
             wc = query_data.get_workload_snapshot_delete_status(
                 tvaultconf.snapshot_name, tvaultconf.snapshot_type_full, snapshot_id)
             LOG.debug("Snapshot Delete status: " + str(wc))
-            if (str(wc) == "1"):
+            while (str(wc) == "0" and (time.time() - start_time < 600)):
+                wc = query_data.get_workload_snapshot_delete_status(
```

Comment:
I have one query here (Its not an issue, just wanted to understand if we can remove the repetitive calls here)- 

In this new code change , now call to "get_workload_snapshot_delete_status" is given twice. One at line no - 217 and second time called at line no - 221 inside while loop.
After calling get_workload_snapshot_delete_status for the first time(line-217), we are not adding any sleep and immediately moving to while() condition. 
If by that time "wc" value returned other than "0" then it will not go in loop. Otherwise it will call get_workload_snapshot_delete_status again until wc value is changed or timeout occur. 

Instead we can call it once, in a while loop directly like,
'''
remove call to get_workload_snapshot_delete_status() at line number 217
set wc = 0 and then iterate in while loop to get snapshot delete status. 
break the loop if wc is other than zero or timeout.
then rest of the code will remain same. 
'''